pkg/gadgets/seccomp/gadget.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seccomp
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"sort"
+	"sync"
+	"syscall"
+
+	"github.com/cilium/ebpf"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	libseccomp "github.com/seccomp/libseccomp-golang"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/types"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+)
+
+// #include "../../../gadget-container/gadgets/seccomp/bpf/common.h"
+import "C"
+
+const (
+	BPF_PROG_NAME = "tracepoint__raw_syscalls__sys_enter"
+	BPF_MAP_NAME  = "syscalls_per_mntns"
+)
+
+type Trace struct {
+	started bool
+	cmd     *exec.Cmd
+	out     bytes.Buffer
+}
+
+type TraceFactory struct {
+	mu     sync.Mutex
+	traces map[string]*Trace
+}
+
+func (f *TraceFactory) LookupOrCreate(name types.NamespacedName) gadgets.Trace {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	if f.traces == nil {
+		f.traces = make(map[string]*Trace)
+	}
+	trace, ok := f.traces[name.String()]
+	if ok {
+		return trace
+	}
+	trace = &Trace{}
+	f.traces[name.String()] = trace
+
+	return trace
+}
+
+func (f *TraceFactory) Delete(name types.NamespacedName) error {
+	log.Infof("Deleting %s", name.String())
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	t, ok := f.traces[name.String()]
+	if !ok {
+		log.Infof("Deleting %s: does not exist", name.String())
```

Comment:
So, in this case the log will print:
Deleting namespace
Deleting namespace: does not exist

Is that repetition intentional? It seems like a weird log entry to me. I'd usually expect a "couldn't delete" or similar message.